### PR TITLE
backend/enh/docker-push-for-ci

### PR DESCRIPTION
**Pull request by vsaimanohar**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables

## Checklist

- [X] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: .github/workflows/nix.yml [hunk 1]
```diff
@@ -26,3 +23,41 @@ jobs:
 
       - name: Build all flake outputs
         run: om ci
+
+      - name: Build and push Docker image
+        if: github.event_name == 'push' && github.event.pull_request.merged == true
```
Comment (srid) [inline .github/workflows/nix.yml:28]: Is there a particular reason for this condition?

You can also check on the branch name, right?

See https://github.com/juspay/omnix/blob/1f62a25f92c84ce88029783ad5a1bad1245edec6/.github/workflows/ci.yaml#L11

File: .github/workflows/nix.yml [hunk 2]
```diff
@@ -26,3 +23,41 @@ jobs:
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -euo pipefail
+
+          ARG1="${1:-dockerImage}"
```
Comment (srid) [inline .github/workflows/nix.yml:34]: Just inline `ARG1`; this is not a script that someone will invoke passing arguments anyway.

File: .github/workflows/nix.yml [hunk 3]
```diff
@@ -26,3 +23,41 @@ jobs:
...
+          shift 1 || true
+
+          set -x
+          # Show git status as Docker tag is based on the working copy status
+          git status
+
+          # Build the flake output and load the Docker image
+          docker load -i "$(nix build .#"$ARG1" --print-out-paths --no-update-lock-file)"
```
Comment (srid) [inline .github/workflows/nix.yml:42]: Inline `ARG1` here1

File: .github/workflows/nix.yml [hunk 4]
```diff
@@ -26,3 +23,41 @@ jobs:
...
+
+          # Fetch the image name and tag
+          IMAGE_NAME="$(nix eval --json .#packages.x86_64-linux."$ARG1".buildArgs | jq -r '"\(.name):\(.tag)"')"
```
Comment (srid) [inline .github/workflows/nix.yml:45]: Inline `ARG1` here as well.

File: .github/workflows/nix.yml [hunk 5]
```diff
@@ -26,3 +23,41 @@ jobs:
...
+
+          # Push the Docker image
+          echo "$GITHUB_TOKEN" | docker login ghcr.io -u "${{ github.actor }}" --password-stdin
+          docker tag "ghcr.io/nammayatri/beckn-gateway:a7c614" "$IMAGE_NAME"
+          docker push "$IMAGE_NAME"
+
+          echo "Pushed image: ${IMAGE_NAME}"
+
+          # Log out from Docker registry
+          docker logout ghcr.io
```
Comment (srid) [inline .github/workflows/nix.yml:63]: Ah, I just realized something. I think we should do the entire docker login and push from a `ubuntu-latest` runner, so we don't have to deal with these security concerns. You can use `actions/upload-artifact` and `actions/download-artifact` - have Nix upload the image tarball, and download it from a ubuntu-latest runner, loading and pushing it.

See 

https://github.com/juspay/omnix/blob/1f62a25f92c84ce88029783ad5a1bad1245edec6/.github/workflows/ci.yaml#L66

https://github.com/juspay/omnix/blob/1f62a25f92c84ce88029783ad5a1bad1245edec6/.github/workflows/ci.yaml#L95

**Review by srid — CHANGES_REQUESTED** [review]

File: .github/workflows/nix.yml (hunk 5 above)
Comment (vsaimanohar) [reply .github/workflows/nix.yml]: hey @srid made changes regarding this

File: .github/workflows/nix.yml [hunk 6]
```diff
@@ -1,28 +1,56 @@
...
-          skipPush: true
 
       - name: Build all flake outputs
         run: om ci
+
+      - name: Docker tasks
+        if: github.ref == 'refs/heads/main'
+        run: |
+          nix build .#dockerImage
+          cp result image.tar
```
Comment (srid) [inline .github/workflows/nix.yml:30]:
```suggestion
          nix build .#dockerImage -o image.tar
```

**Review by srid — CHANGES_REQUESTED** [review]
Looks good! Just needs these two changes.

**Comment by srid** [comment]
@vsaimanohar Can you also delete `Jenkinsfile` in this branch? So we officially migrate this repo to Github Actions as this PR is merged.

**Review by srid — APPROVED** [review]

**Pull request merged**